### ISSUE TITLE
Thunderbolt and Sleep

### ISSUE THREAD

**Issue by seargentpepper** [pierpaolodimarzo/ThinkPad-T480#21]
Thanks Pier for your pre-built, the starting on with the macOS sound it's really accurate. The EFI works incredible, in fluidity and responsiveness. I'm still adapting to this model T480.

Sadly, I'm facing some issues with Thunderbolt (not critical)

It just doesn't show up, but I'm not concerned as I don't need to plug a Thunder device.

But i had been trying to fix the sleep as you recommended, and its no being posible. Any help it would be appreciated.

About the TB I followed you guide but erased windows partition as I don't have any plans to use it.

About the sleep, as most users I need to close my notebook constantly while I'm moving around different buildings. Every time I shutdown or sleep MacOS, it thinks the system was hard-rebooted. 

 ![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/de5942b6-1c68-432f-8f75-f1022b1a3c9a)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/047c4fa5-f7b1-42e8-872b-6431e72b4e37)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/f3a739da-8518-49e4-a41a-49e90e5301e8)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/66fe9bc4-1578-4cc1-8bf2-a9c036f4f0dd)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/9e56396e-6ea6-49a8-9fd3-80c389a90a75)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/95310bb4-14dd-4e7f-94fa-148b161c442d)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/2472bcda-73b4-42b6-9e65-63598245a41a)
![image](https://github.com/pierpaolodimarzo/ThinkPad-T480/assets/20083347/575b928c-d216-4f03-88d3-b386860e0bde)

**Comment by bweh** [comment]
Hey, I had the same issue with thunderbolt not showing up despite being on NVM version 23.0. I randomly got it to work for now by setting `Support in Pre Boot Environment > Thunderbolt(TM) device` to disabled. Not sure if it's ideal, but can you try it?

**Comment by Mifaxav** [comment]
Yeah that's also how i got TB to work with mine. Im currently on sonoma RC (installed using iMac19,2 smbios due to T2 error) and have my thunderbolt working but my sleep isn't working. It's rebooting as a wake the machine up. One other thing i have observed though is that if i change my smbios to any other machine that's compatible but not in the MacBookPro15,x lineup, sleep is perfect but my camera, bluetooth and touchscreen go for a toss. Btw im on my own EFI which is on my repo.. Rlly ned help with this.

Thanks,
